tracing-core/src/span.rs

```diff
@@ -181,3 +206,75 @@ impl<'a> Record<'a> {
         self.values.is_empty()
     }
 }
+
+// ===== impl Current =====
+
+impl Current {
+    /// Constructs a new `Current` that indicates the current context is a span
+    /// with the given `metadata` and `metadata`
+    pub fn new(id: Id, metadata: &'static Metadata<'static>) -> Self {
+        Self {
+            inner: CurrentInner::Current { id, metadata },
+        }
+    }
+
+    /// Constructs a new `Current` that indicates the current context is *not*
+    /// in a span.
+    pub fn none() -> Self {
+        Self {
+            inner: CurrentInner::None,
+        }
+    }
+
+    /// Constructs a new `Current` that indicates the `Subscriber` does not
+    /// track a current span.
+    pub(crate) fn unknown() -> Self {
+        Self {
+            inner: CurrentInner::Unknown,
+        }
+    }
+
+    /// Returns `true` if the `Subscriber` that constructed this tracks a
```

Comment:
I'm thinking this could be made a bit clearer. How about:

```suggestion
    /// Returns `true` if the `Subscriber` that constructed this `Current` tracks a current span.
```